### ISSUE TITLE
make cache key function names more specific

### ISSUE THREAD

**Issue by jeromedockes** [probabl-ai/skore#3096]
the second item of the cache key  is informally "the function name" but nothing is enforced. it would help if they were all at least prefixed with the accessor name (if any) to avoid name clashes and make it easier to inspect

**Comment by auguste-probabl** [comment]
A motivating example is that defining a metric with the same name as an inspection method can make cache keys ambiguous (although there is currently no bug related to this):
```py
>>> from skore import evaluate
>>> from sklearn.linear_model import Ridge
>>> import sklearn
>>> X, y = sklearn.datasets.fetch_california_housing(return_X_y=True, as_frame=True)
>>> model = Ridge()
>>> report = evaluate(model, X, y)
>>> report.inspection.permutation_importance()
>>> print(list(report._cache.keys()))
# [
# ('test', 'predict', None),
# ('test', 'permutation_importance', ('mapping', (('at_step', 0), ('max_samples', 1.0), ('metric', None), ('n_repeats', 5), ('seed', None))))
# ]
 
>>> report.metrics.add(lambda e,X,y: 1, name="permutation_importance")
>>> report.metrics.summarize(metric="permutation_importance")
>>> print(list(report._cache.keys()))
# [
# ('test', 'predict', None),
# ('test', 'permutation_importance', ('mapping', (('at_step', 0), ('max_samples', 1.0), ('metric', None), ('n_repeats', 5), ('seed', None)))),
# ('test', 'permutation_importance', ('mapping', ()))  # Nothing to differentiate the two keys
# ]
```
